src/standard/image.rs

```diff
@@ -669,6 +704,85 @@ impl<E: OclPrm> Image<E> {
         Ok(new_img)
     }
 
+    /// Returns a new `Image` from an existant GL texture2D/3D.
+    // [WORK IN PROGRESS]
+    pub fn from_gl_texture(queue: &Queue, flags: MemFlags, image_desc: ImageDescriptor)
+            -> OclResult<Image<E>>
+    {
+        // FIXME need these defines !
+        // https://www.khronos.org/registry/cl/specs/opencl-1.x-latest.pdf#page=280
+        let texture_target = 0;
+        let miplevel = 0;
+        let texture = 0;
+
+        let obj_core = match image_desc.image_depth {
+            2 => unsafe { try!(core::create_from_gl_texture_2d(
+                                queue.context_core_as_ref(),
+                                texture_target,
+                                miplevel,
+                                texture,
+                                flags)) },
+            3 => unsafe { try!(core::create_from_gl_texture_3d(
```

Comment:
Ok so after looking at this some more, I've noticed that `clCreateFromGLTexture2D` and `clCreateFromGLTexture3D` are deprecated functions since 1.1. 

In the future I want compatibility with all versions of OpenCL but for now I want to focus on version 1.2. Would you mind please re-wiring this up to use [clCreateFromGLTexture](https://www.khronos.org/registry/cl/sdk/1.2/docs/man/xhtml/clCreateFromGLTexture.html) instead? There is already a commented out signature in the `cl_gl_h.rs` file.

Also please let me know if you need help implementing these extra OpenGL enums like `cl_gl_object_type` etc. I think you can figure it out but let me know if you want me to do it.